bindings/matrix-sdk-ffi/src/room.rs

```diff
@@ -378,6 +378,28 @@ impl Room {
         Ok(())
     }
 
+    /// Send a raw event to the room.
+    ///
+    /// # Arguments
+    ///
+    /// * `event_type` - The type of the event to send.
+    ///
+    /// * `content` - The content of the event to send as JSON string.
+    pub async fn send_custom_event(
+        &self,
+        event_type: String,
+        content: String,
+    ) -> Result<(), ClientError> {
+        let content_json: serde_json::Value = match serde_json::from_str(&content) {
+            Ok(json) => json,
+            Err(e) => {
+                return Err(ClientError::Generic { msg: format!("Failed to parse JSON: {}", e) })
+            }
+        };
+        self.inner.send_raw(&event_type, content_json).await?;
+        Ok(())
```

Comment:
This can be more concisely expressed using `map_err()` to convert the error type, a bit of whitespace also doesn't hurt:

```suggestion
       let content_json: serde_json::Value = serde_json::from_str(&content)
            .map_err(|e| ClientError::Generic { msg: format!("Failed to parse JSON: {e}") })?;

        self.inner.send_raw(&event_type, content_json).await?;

        Ok(())
```